docs/cadence/language/access-control.md

```diff
@@ -369,13 +369,7 @@ resource SubResource {
 }
 
 resource OuterResource {
-    access(self) let childResource: @SubResource
-
-    // by referering to `Map` here, we declare that the entitlements we receive when accessing the `getRef` function on this resource
-    // will depend on the entitlements we possess to the resource during the access. 
-    access(Map) fun getRef(): auth(Map) &SubResource {
-        return &self.childResource as auth(Map) &SubResource
-    }
+    access(Map) let childResource: @SubResource
```

Comment:
I did not specifically add about resource typed fields, since this section already describes how entitlement mappings pass down the entitlements in general. Also, the newly added `Field and Index Access` (in reference.md) section describes how references and entitlements are received when accessing fields through a reference.

If we want to have a more granular/in-depth explanation on how entitlement mappings behave on each different use case (functions, references, and container-typed field) separately, then I guess that also makes sense and would be good to have it too. But probably unrelated to this change, so maybe we could do it in a separate PR.